File: programming files/sum_even_fibonacci.cc
//while b > 40000000 instead of < in line 9 - CC

#include <iostream>

int main()
{
    int a = 0, b = 1;
    int sum_even = 0;
    // FIXED: Turned > to < in while loop WB
    while (b < 4000000) {
        if (b % 2 == 0) {
            sum_even += b;
        }
        int temp = b;
        b = a + b;
        a = temp;
    }
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732
// Did you check your work Harry?

File: programming files/sum_even_fibonacci.py
# Bugs introduced: changed b % 2 to b / 2, changed a, b = b, a+b to a, b = b, a-b  WB
# declare variables a and b
a, b = 0, 1
sum_even = 0
while b < 4000000:
    # if b is even 
    if b / 2 == 0:
        sum_even += b
    a, b = b, a-b

#print the even sum
print(sum_even)

# answer should be 4613732
